chapter09/axp/ex01

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/perl
+use utf8;
+use strict;
+use warnings;
+use List::Util qw( reduce );
+
+use v5.12;
+
+my %patterns = (
+  Gilligan   => qr/(?:Wiley )?Gilligan/,
+  'Mary-Ann' => qr/Mary-Ann/,
+  Ginger     => qr/Ginger/,
+  Professor  => qr/(?:The )?Professor/,
+  Skipper    => qr/Skipper/,
+  'A Howell' => qr/Mrs?. Howell/,
+);
+
+say rightmost(
+	'There is Mrs. Howell, Ginger, and Gilligan',
+	\%patterns
+);	
+	
+sub rightmost {
+	my ( $string, $patterns ) = @_;
+
+	return (
+		reduce {
+			my $b_idx = $string =~ m/$patterns->{$b}/ ? $-[0] : -1;
```

Comment:
It was in the chapter when talking about the rightmost function. Who thinks @thatguy59 didn't read the chapter?